src/Model/EpisodeManager.php

```diff
@@ -0,0 +1,73 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: aragorn
+ * Date: 19/04/18
+ * Time: 15:41
+ */
+
+namespace Model;
+
+
+class EpisodeManager extends AbstractManager
+{
+    const TABLE = 'episode';
+
+    /**
+     *  Initializes this class.
+     */
+    public function __construct ()
+    {
+        parent::__construct(self::TABLE);
+    }
+
+    /**
+     * @param array $data
+     * @return string|void
+     * @throws \Exception
+     */
+    public function insert (array $data)
+    {
+        $idSerie = $data[ 'idSerie' ];
+        $nb = $data[ 'numeroSeason' ];
+
+        // On récupère l'id de la saison associée à la série
+
+        $query = "SELECT id FROM season WHERE number_season = :nb AND idserie = :IdSerie";
+        $statement = $this->pdoConnection->prepare($query);
+        $statement->setFetchMode(\PDO::FETCH_CLASS, $this->className);
+        $statement->bindValue('nb', $nb);
+        $statement->bindValue('IdSerie', $idSerie);
+        $statement->execute();
+        $idSeason = $statement->fetch()->getId();
+
+        //On vérifie que l'épisode n'existe pas déjà
+
+        $verif = "SELECT count(number) as count FROM $this->table WHERE idseason = :idSeason AND idserie = :idSerie AND number = :number";
+        $result = $this->pdoConnection->prepare($verif);
+        $result->setFetchMode(\PDO::FETCH_ASSOC);
+        $result->bindValue('idSeason', $idSeason, \PDO::PARAM_INT);
+        $result->bindValue('idSerie', $idSerie, \PDO::PARAM_INT);
+        $result->bindValue('number', $data[ 'numeroEpisode' ], \PDO::PARAM_INT);
+        $result->execute();
+        $res = $result->fetchAll();
+        $count = $res[0]['count'];
+
+        if ($count !== '0') {
+            throw new \Exception('L\' épisode existe déjà');
```

Comment:
tout le code vérifiant si un episode existe déjà devrait etre mis dans méthode $this->checkSeasonExist() qui te renvoie true ou false (en fonction de $count==0 ou non)
ça simplifierait grandement ton code. De plus la "validation" ne devrait pas etre dans ton manager, mais plutot dans une classe Validator, ou sinon dans ton controleur. 
Dans tous les cas, quand tu as une fonction aussi longue, généralement c'est que tu peux refactoriser